raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -167,16 +229,35 @@ public boolean updateLocalEndOffset(LogOffsetMetadata endOffsetMetadata) {
 
     private static class ReplicaState implements Comparable<ReplicaState> {
         final int nodeId;
-        boolean hasEndorsedLeader;
+
         Optional<LogOffsetMetadata> endOffset;
+
         OptionalLong lastFetchTimestamp;
+        long lastCaughtUpTimeMs = 0L;
```

Comment:
We should initialize to -1 as well, to indicate "unknown", since in the code we have `lastFetchTimestamp = OptionalLong.of(Math.max(lastFetchTimestamp.orElse(-1L), currentFetchTimeMs));` so we may actually go backwards from 0 to -1.